internal/pkg/template/templates/workloads/services/backend/cf.yml

```diff
@@ -102,7 +102,7 @@ Resources:
       {{- end}}
     Properties:
       {{- "\n"}}{{ include "service-base-properties" . | indent 6 }}
-      ServiceRegistries: !If [ExposePort, [{RegistryArn: !GetAtt DiscoveryService.Arn, Port: !Ref ContainerPort}], !Ref "AWS::NoValue"]
+      ServiceRegistries: !If [ExposePort, [{RegistryArn: !GetAtt DiscoveryService.Arn, Port: !Ref TargetPort}], !Ref "AWS::NoValue"]
```

Comment:
Actually this is not working. `ExposePort` is for main container, and we should check targetPort instead.

Reply:
good catch. changed backend service params to always include target port, and changed the `ExposePort` condition to use target port.